File: content/0-basics/2-hello-world/langgraph/3-agent_gemini.py
from langchain.agents import create_agent
from langchain_google_genai import ChatGoogleGenerativeAI

LLM_MODEL = "gemini-pro-latest"

model = ChatGoogleGenerativeAI(model=LLM_MODEL)

# create_agent builds a graph that loops: LLM → tools → LLM
# with no tools, it's a simple prompt → response
agent = create_agent(model, tools=[])

result = agent.invoke({
    "messages": [("user", "What is the capital of France?")]
})
print(result["messages"][-1].text)
# "Paris."

File: content/0-basics/2-hello-world/aisdk/3-agent_gemini.ts
import { generateText } from "ai";
import { google } from "@ai-sdk/google";

const LLM_MODEL = "gemini-pro-latest";

const { text } = await generateText({
  model: google(LLM_MODEL),
  prompt: "What is the capital of France?",
});
console.log(text);
// "Paris."
